lib/page-object/element.js

```diff
@@ -1,24 +1,169 @@
 /**
- * Class that all elements subclass from
+ * Class that all elements subclass from.
  *
- * @param {Object} options Element options defined in page object
+ * @param {Object} definition User-defined element options defined in page object.
+ * @param {Object} options Additional options to be given to the element.
  * @constructor
  */
-function Element(options) {
-  this.parent = options.parent;
+function Element(definition, options) {
+  options = options || {};
+
+  this.name = options.name;
 
-  if (!options.selector) {
-    throw new Error('No selector property for element "' + options.name +
-      '" Instead found properties: ' + Object.keys(options));
+  if (!definition.selector) {
+    throw new Error('No selector property for ' + this.constructor.name.toLowerCase() + ' "' + getName(this) +
```

Comment:
I think it would be a bit clearer if we extract the following variable:

```js
var classType = this.constructor.name.toLowerCase(); // either Element or Section
```